lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -73,6 +73,85 @@ def list_all(filter = {})
         filter.empty? ? results : results.select { |obj| filter.all? { |k, v| obj.public_send(k) == v } }
       end
 
+      # This method returns a model object based on an ID string for a Service.
+      #
+      # Example:
+      #
+      #   vms = Azure::Armrest::VirtualMachineService.new(conf)
+      #
+      #   vm = vms.get('your_vm', 'your_group')
+      #   nic_id = vm.properties.network_profile.network_interfaces[0].id
+      #   nic = vm.get_by_id(nic_id)
+      #
+      def get_by_id(id_string)
```

Comment:
Hm, ok, but I've kept the originals as aliases because I like short method names. :)